client-ui/src/app/model/lineage-graph/leneage-graph-legend.models.ts

```diff
@@ -15,26 +15,31 @@
  */
 
 
-import { LINE_COLOR_HLT_IMPACT, LINE_COLOR_HLT_LINEAGE, LINE_COLOR_HLT_PRIMARY } from './leneage-graph.models';
+import {LINE_COLOR_HLT_IMPACT, LINE_COLOR_HLT_LINEAGE, LINE_COLOR_HLT_PRIMARY} from './leneage-graph.models';
 
 
-export interface LineageGraphLegend {
+export type LineageGraphLegend = {
```

Comment:
Why it should be a type over the rule "interface-over-type-literal"?

Reply:
I'm pretty sure this rule should be considered outdated. There are many thoughts and debates about types vs interfaces, why both exist in the same language and what to use when, in most cases it's the matter of taste (unfortunately). But I'm in the campus that say that it's the matter of paradigm in use, rather then just taste.
In modern TS there aren't many things that interfaces could do and types are not and vice versa (event in fact the types could do much more things over interfaces then the opposite). 

The ___most important___ difference however is not what they can or cannot do, but what information (the purpose) they communicate to the developer. 

The interfaces as well as classes originate in the Object-Oriented paradigm. Interfaces describe the object contract and are meant to be implemented by some class. Even though it's possible to describe the data structure with an interface, the main focus is usually made on behavioral aspect - the methods.

The types on the other hand is a purely functional  beast. It is made to describe data types in all possible aspects. Types cover any types, not only object ones and are very flexible.

Do the general rule of thumb as I see it is simple - if you aim to implement X in some class in your project and/or X is logically more about behavior then structure then X is an interface. Otherwise X is a type.